### ISSUE TITLE
License plate with two lines of text problem

### ISSUE THREAD

**Issue by vinhpt-ueh** [ankandrew/fast-plate-ocr#39]
I have experimented with Brazilian plates that have two lines of text. The model is able to detect characters; however, they are mixed together and not in the correct order.
How can we improve the model?

![Image](https://github.com/user-attachments/assets/539cd236-ebe2-4fb5-b413-84955daf2a67)

**Comment by ankandrew** [comment]
Hi, this might be because some countries have that kind of plate, i.e. plate is `KA4902MM`:

![Image](https://github.com/user-attachments/assets/48db07e0-7663-408a-89f6-41d5289d0dea)

The model might be thinking that this is one of those cases. But in your shared image, isn't `BEE4R22` and `BRA20E4` the correct plate text order? Running it with global model in demo https://huggingface.co/spaces/ankandrew/fast-alpr.

![Image](https://github.com/user-attachments/assets/94a13459-1b4b-4abd-b083-131549281879)

**Comment by vinhpt-ueh** [comment]
I apologize for using a misleading image, it is only used to represent a 2-line license plate. The actual image I used will be attached here. I have tried the available pre-trained models and have not achieved the expected results. I am trying to finetune the model.
Below is the predicted result for 2-line plate.
![Image](https://github.com/user-attachments/assets/5bcab0e1-b5af-4f1c-9378-d77f365913da)
Some 2-line plates from the data:

![Image](https://github.com/user-attachments/assets/3642782e-628e-4219-b411-d0165abf256d)

![Image](https://github.com/user-attachments/assets/27e5efd0-ead8-4a0c-804a-1f519f58eb0d)

![Image](https://github.com/user-attachments/assets/6e770abc-4e57-4209-9943-41f057ad5612)

![Image](https://github.com/user-attachments/assets/6db5f645-2bc8-4638-ae3f-9ea11aa83122)

![Image](https://github.com/user-attachments/assets/6a9fb4c9-9bf9-4aa9-80e2-3ed816401d92)

![Image](https://github.com/user-attachments/assets/b564507f-83c1-4341-b440-d478e2f0f0c7)

![Image](https://github.com/user-attachments/assets/b1b99fc7-7dfa-4716-982a-6cd9c59ccafe)

![Image](https://github.com/user-attachments/assets/f13beb52-2174-4b70-9766-c134fe759016)

![Image](https://github.com/user-attachments/assets/2ecab378-b1be-4dab-95b4-d1044cdb2088)

![Image](https://github.com/user-attachments/assets/74666e5c-9a5b-43e5-bbd7-c2065ff841cc)

**Comment by hungtrieu07** [comment]
Hello @ankandrew, I have 2 questions: 
- If my own dataset have 2 types of license plate: single line and double line look like this 2 images, how can I label them? In my opinion, the first image is 77B024.71, the second image is 79A-085.99.
- With the dataset have dash (-) and dot (.) character, which way should I choose: train from scratch or finetine the model?

![Image](https://github.com/user-attachments/assets/2a78431f-30c8-44a7-95a1-1f40efdd1d60)
![Image](https://github.com/user-attachments/assets/deca66ba-027c-4dfd-ba3a-684037e9c5cc)

**Comment by ankandrew** [comment]
hey @hungtrieu07!

1. Yes, personally I'd label them the same way. It actually doesn't matter your labeling schema, the only important thing is that you are **consistent** on how you label them, the model will learn it accordingly.
2. Try fine-tuning first, the models have seen a lot of plates (even though it hasn't learned in the head to predict `-` or `.`) but many of those learned features are probably useful.

**Comment by hungtrieu07** [comment]
> hey [@hungtrieu07](https://github.com/hungtrieu07)!
> 
> 1. Yes, personally I'd label them the same way. It actually doesn't matter your labeling schema, the only important thing is that you are **consistent** on how you label them, the model will learn it accordingly.
> 2. Try fine-tuning first, the models have seen a lot of plates (even though it hasn't learned in the head to predict `-` or `.`) but many of those learned features are probably useful.

I will try that, thanks a lot!
